test-app/test/JavaControllerTest.java

```diff
@@ -92,7 +86,7 @@ private TestUsernamePasswordAuthProvider upAuthProvider() {
 				.plugin(TestUsernamePasswordAuthProvider.class);
 	}
 
-	private UserServicePlugin userServicePlugin() {
+	private AbstractUserService userServicePlugin() {
 		return Play.application().plugin(TestUserServicePlugin.class);
```

Comment:
this should be `Play.application().injector().instanceOf(TestUserServicePlugin.class)` and even then the `Play.application()` method is deprecated so we probably need to make this class extend WithApplication class so app can be provided for us. You get it right?